screens/RegisterScreen.js

```diff
@@ -0,0 +1,181 @@
+                error:'',
+                loading: false
+            }
+        }
+        this._handleAdd = this._handleAdd.bind(this);
+        this.onRegistrationFail = this.onRegistrationFail.bind(this);
+    }
+
+    componentWillUnmount() {
+        this.setState = {
+            value: {
+                name: '',
+                email: '',
+                username:'',
+                password: null,
+            }
+        }
+    }
+
+    _onChange = (value) => {
+        this.setState({
+            value
+        })
+    }
+
+    _handleAdd = () => {
+        const value = this.refs.form.getValue();
+        //IF the form valid ..
+        this.setState({ error: '', loading: true });
+        if (value) {
+            const data = {
+                name: value.name,
+                email: value.email,
+                username: value.username,
+                password: value.password,
+            }
+            let payload = []
+            for (let property in data) {
+                let encodedKey = encodeURIComponent(property)
+                let encodedValue = encodeURIComponent(data[property])
+                payload.push(encodedKey + "=" + encodedValue)
+            }
+    payload = payload.join("&")
+    console.log(`payload: ${payload}`)
+    //sent post request
+    fetch('http://192.168.100.14:8000/v1/paylist/user/signup', {
+    method: 'POST',
+    headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
+        Accept : 'application/x-www-form-urlencoded'
+    },
+    body: payload
+    })
+    .then(res => {
+    resStatus = res.status
+    return res.json()
+    })
+    .then(res => {
+        switch (resStatus) {
+          case 200:
+            console.log('success')
+            alert('You may login now')
+            break
+        //   case 400:
+        //     if (res.code === 'ValidationFailed') {
+        //     // My custom error messages from the API.
+        //         console.log('field can not be null')
+        //         alert('field can not be null')
+        //     }
+        //     break
+          case 500:
+            console.log('username exist')
+            alert('username exist')
+            break
+          default:
+            console.log('unhandled')
+            break
+        }
+      })
+        .done()
+        } else {
+            //form validation error
+            alert('Please fill the empty field')
+        }
+    }
+
+    onRegistrationFail() {
+        this.setState({
+            error: 'Registration Failed',
+            loading: false
+        });
+    }
+
+    render() {
+        return (
+            <ScrollView style={styles.container}>
+                <Form ref='form' type={newUser} options={option}
+                value={this.state.value} onChange={this._onChange}/>
+                <TouchableHighlight onPress={this._handleAdd}>
+                    <Text style={[styles.button, styles.greenButton]}>Create Account</Text>
```

Comment:
@offler9 why we dont use `button` from react native paper?
https://callstack.github.io/react-native-paper/button.html